src/codec-protobuf.js

```diff
@@ -49,33 +58,29 @@ class Codec {
 
     if (this._verify) {
       const err = type.verify(message);
-
       if (err) {
-        throw new Error(`CodecProtobuf: Verify error by ${err}`);
+        throw new Error(`invalid object [${err}]`);
       }
     }
 
-    const value = type.encode(message).finish();
-    return AnyType.encode({ type: typeName, value }).finish();
+    return Any.encode({
+      type: typeName,
+      value: type.encode(message).finish()
+    }).finish();
   }
 
   decode (buffer, withType = this._decodeWithType) {
-    const obj = this._decode(buffer);
-
-    if (withType === true) {
-      return obj;
-    }
-
-    return obj.message;
-  }
-
-  _decode (buffer) {
-    const { type: typeName, value } = AnyType.toObject(AnyType.decode(buffer));
+    const { type: typeName, value } = Any.toObject(Any.decode(buffer));
 
     const type = this._root.lookupType(typeName);
     const message = type.toObject(type.decode(value));
 
-    return { type: typeName, message };
+    // TODO(burdon): Must not have functions that have different return types based on params!
```

Comment:
Agree but this should be something to discuss because we are using this flag in many parts of wireline-core.